KVStore corrupts binary data

I'm trying to store binary data in the KV, but the data appears to getting corrupted.  Consider this code:

```rust
.get_async("/test", |req, ctx| async move {
            let kv = ctx.kv("UPLOAD")?;
            let bytes = [0xd2, 0xda, 0x1e, 0x77, 0x3c, 0x50, 0xd9, 0x35, 0x9a, 0x89, 0xd5, 0x53, 0xc3, 0x81, 0x6d, 0x20];
            kv.put_bytes("test", &bytes)?.execute().await?;

            let bytes_from_kv = kv.get("test").await?.unwrap();
            assert_eq!(&bytes, bytes_from_kv.as_bytes());

            Response::empty()
        })
```

Trying to run this yields this assertion failure:

```
panicked at 'assertion failed: `(left == right)`
  left: `[210, 218, 30, 119, 60, 80, 217, 53, 154, 137, 213, 83, 195, 129, 109, 32]`,
 right: `[239, 191, 189, 239, 191, 189, 30, 119, 60, 80, 239, 191, 189, 53, 239, 191, 189, 239, 191, 189, 239, 191, 189, 83, 195, 129, 109, 32]`', src/lib.rs:56:13
```

However when I download this value via the https://dash.cloudflare.com/, the data is correct (so it seems like the corruption is happening during the fetch on the rust side)

## Version info
* worker-rs version v0.0.7
* wrangler version 1.19.6
* rustc version 1.59.0-nightly (7abab1efb 2021-12-17)

I confirm your latest version works @zebp , thanks.  Unfortunately mixing in your version is not ideal since it's not possible to use the question-mark operator to propagate errors:

```
error[E0277]: the trait bound `worker::Error: From<worker_kv::KvError>` is not satisfied
the trait `From<worker_kv::KvError>` is not implemented for `worker::Error`
```

So I'm definitely looking forward to seeing this fully integrated into the next `workers-rs` version 👍

--- a/README.md
+++ b/README.md
@@ -64,8 +64,8 @@ pub async fn main(req: Request, env: Env) -> Result<Response> {
         .get_async("/account/:id", |_req, ctx| async move {
             if let Some(id) = ctx.param("id") {
                 let accounts = ctx.kv("ACCOUNTS")?;
-                return match accounts.get(id).await? {
-                    Some(account) => Response::from_json(&account.as_json::<Account>()?),
+                return match accounts.get(id).json::<Account>().await? {
+                    Some(account) => Response::from_json(&account),
                     None => Response::error("Not found", 404),
                 };
             }

--- a/worker-sandbox/src/lib.rs
+++ b/worker-sandbox/src/lib.rs
@@ -180,8 +180,8 @@ pub async fn main(req: Request, env: Env) -> Result<Response> {
         .get_async("/formdata-file-size/:hash", |_, ctx| async move {
             if let Some(hash) = ctx.param("hash") {
                 let kv = ctx.kv("FILE_SIZES")?;
-                return match kv.get(&hash).await? {
-                    Some(val) => Response::from_json(&val.as_json::<FileSize>()?),
+                return match kv.get(&hash).json::<FileSize>().await? {
+                    Some(val) => Response::from_json(&val),
                     None => Response::error("Not Found", 404),
                 };
             }

--- a/worker/Cargo.toml
+++ b/worker/Cargo.toml
@@ -24,6 +24,6 @@ serde_json = "1.0.64"
 url = "2.2.2"
 wasm-bindgen = "0.2.78"
 wasm-bindgen-futures = "0.4.28"
-worker-kv = "0.4.0"
+worker-kv = "0.5.0"
 worker-macros = { path = "../worker-macros", version = "0.0.2" }
 worker-sys = { path = "../worker-sys", version = "0.0.2" }

